rest/route/pod_agent.go

```diff
@@ -304,8 +304,17 @@ func (h *podAgentNextTask) Run(ctx context.Context) gimlet.Responder {
 	})
 }
 
-// enqueuePodTerminationJob will enqueue the job to terminate a pod with a detailed reason for doing so.
-func (h *podAgentNextTask) enqueuePodTerminationJob(ctx context.Context, reason string) error {
+// prepareForPodTermination will mark the pod as preparing to terminate if it is
+// not already attempting to terminate and enqueue the job to terminate it with
+// a detailed reason for doing so.
+func (h *podAgentNextTask) prepareForPodTermination(ctx context.Context, p *pod.Pod, reason string) error {
+	if p.Status != pod.StatusDecommissioned && p.Status != pod.StatusTerminated {
+		// kim: TODO: test
```

Comment:
If this is being implemented later can we ticket and link it here?

Reply:
Ah, this is already tested, I just forgot to remove the note to myself to test it 🤦 . Should be good now.